src/collection-preferences/content-display/index.tsx

```diff
@@ -51,7 +58,13 @@ export default function ContentDisplayPreference({
   const titleId = `${idPrefix}-title`;
   const descriptionId = `${idPrefix}-description`;
 
-  const sortedOptions = getSortedOptions({ options, contentDisplay: value });
+  const sortedOptions = useMemo(
+    () =>
+      getSortedOptions({ options, contentDisplay: value }).filter(option =>
+        option.label.toLowerCase().includes(columnFilteringText.toLowerCase())
```

Comment:
Should we trim filtering text when checking? Also in the condition below: `disabled={columnFilteringText.length > 0}`